src/libraries/Common/src/System/Security/Cryptography/MLDsaImplementation.Windows.cs

```diff
@@ -2,47 +2,216 @@
+
+            const string BlobType = Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PUBLIC_BLOB;
+
+            SafeBCryptKeyHandle key =
+                PqcBlobHelpers.EncodeMLDsaBlob(
+                    PqcBlobHelpers.GetParameterSet(algorithm),
+                    source,
+                    BlobType,
+                    static blob => Interop.BCrypt.BCryptImportKeyPair(s_algHandle, BlobType, blob));
+
+            return new MLDsaImplementation(algorithm, key, hasSeed: false, hasSecretKey: false);
+        }
+
+        internal static partial MLDsaImplementation ImportSecretKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            Debug.Assert(s_algHandle is not null, $"Check {SupportsAny}() before calling.");
+
+            const string BlobType = Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PRIVATE_BLOB;
+
+            SafeBCryptKeyHandle key =
+                PqcBlobHelpers.EncodeMLDsaBlob(
+                    PqcBlobHelpers.GetParameterSet(algorithm),
+                    source,
+                    BlobType,
+                    static blob => Interop.BCrypt.BCryptImportKeyPair(s_algHandle, BlobType, blob));
+
+            return new MLDsaImplementation(algorithm, key, hasSeed: false, hasSecretKey: true);
+        }
+
+        internal static partial MLDsaImplementation ImportSeed(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            Debug.Assert(s_algHandle is not null, $"Check {SupportsAny}() before calling.");
+
+            const string BlobType = Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PRIVATE_SEED_BLOB;
+
+            SafeBCryptKeyHandle key =
+                PqcBlobHelpers.EncodeMLDsaBlob(
+                    PqcBlobHelpers.GetParameterSet(algorithm),
+                    source,
+                    BlobType,
+                    static blob => Interop.BCrypt.BCryptImportKeyPair(s_algHandle, BlobType, blob));
+
+            return new MLDsaImplementation(algorithm, key, hasSeed: true, hasSecretKey: true);
+        }
+
+        protected override void ExportMLDsaPublicKeyCore(Span<byte> destination)
+        {
+            ArraySegment<byte> keyBlob = Interop.BCrypt.BCryptExportKey(_key, Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PUBLIC_BLOB);
+
+            try
+            {
+                ReadOnlySpan<byte> keyBytes = PqcBlobHelpers.DecodeMLDsaBlob(
+                    keyBlob,
+                    out ReadOnlySpan<char> parameterSet,
+                    out string blobType);
+
+                Debug.Assert(blobType == Interop.BCrypt.KeyBlobType.BCRYPT_PQDSA_PUBLIC_BLOB);
+
+                // Length is known, but we'll slice just in case
+                MLDsaAlgorithm algorithm = PqcBlobHelpers.GetMLDsaAlgorithmFromParameterSet(parameterSet);
+                Debug.Assert(keyBytes.Length == algorithm.PublicKeySizeInBytes);
+                keyBytes.Slice(0, algorithm.PublicKeySizeInBytes).CopyTo(destination);
```

Comment:
In a release build we don't have the asserts; so if Windows decided to return a value of the wrong size this is now going to fail with an ArgumentException saying that the slice length is too long (if Windows gave us back too little), or we'll succeed and return some unknown diminished slice.

I'd prefer to be paranoid and do it as runtime checks.

```c#
if (blobType != BCRYPT_PQDSA_PUBLIC_BLOB || keyBytes.Length != algorithm.PublicKeySizeInBytes)
{
    Debug.Fail($"blobType: {blobType}, keyBytes.Length: {keyBytes.Length} / {algorithm.PublicKeySizeInBytes}");
    throw new CryptographicException();
}

keyBytes.CopyTo(destination);
```